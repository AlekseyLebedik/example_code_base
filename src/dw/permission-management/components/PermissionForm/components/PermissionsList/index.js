import React from 'react';
import PropTypes from 'prop-types';
import { DragDropContext } from 'react-dnd';
import HTML5Backend from 'react-dnd-html5-backend';

import Tooltip from '@material-ui/core/Tooltip';
import Paper from '@material-ui/core/Paper';
import Fab from '@material-ui/core/Fab';
import Icon from '@material-ui/core/Icon';

import Permission from '../Permission';

import styles from './index.module.css';

const PermissionsList = DragDropContext(HTML5Backend)(
  ({
    contentType,
    fields,
    onMove,
    dragTypeObjectPerm,
    dragTypeDetails,
    form,
  }) => (
    <Paper className={styles.container}>
      <label className={styles.titleContent}>{contentType.model}</label>
      <div>
        {fields.map((selection, index) => (
          <Permission
            key={`${selection}`}
            fields={fields}
            form={form}
            selection={selection}
            contentType={contentType}
            index={index}
            onDrop={item => {
              onMove(item.index, index, fields, selection, item);
            }}
            dragTypeDetails={dragTypeDetails}
            dragTypeObjectPerm={dragTypeObjectPerm}
          />
        ))}
        <Tooltip className={styles.addPermission} title="Add Permission">
          <Fab
            variant="extended"
            onClick={() => {
              fields.push({ permissions: [], selectedDetails: [] });
            }}
            color="primary"
            aria-label="Add"
          >
            <Icon>add</Icon>
            ADD PERMISSION
          </Fab>
        </Tooltip>
      </div>
    </Paper>
  )
);

PermissionsList.propTypes = {
  contentType: PropTypes.object,
  fields: PropTypes.object.isRequired,
  onMove: PropTypes.func.isRequired,
};

PermissionsList.defaultProps = {
  contentType: {},
};

export default PermissionsList;
