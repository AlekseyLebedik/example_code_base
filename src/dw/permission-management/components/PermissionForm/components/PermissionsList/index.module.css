.container {
  padding: 20px;
  margin: 10px 10px 10px 2px;
}

.addPermission {
  margin-top: 10px;
  height: 28px;
}

.titleContent {
  text-transform: capitalize;

  font-weight: bold;
}
