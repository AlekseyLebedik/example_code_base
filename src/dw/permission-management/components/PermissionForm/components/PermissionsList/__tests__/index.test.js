import React from 'react';
import { shallow } from 'enzyme';

import PermissionsList from '../index';

describe('Permissions List component', () => {
  const props = {
    contentType: {
      id: 1,
      model: 'contentType1',
      details: [{ id: 2, str: 'detail1' }],
      permissions: [{ id: 1, name: 'perm1' }],
    },
    fields: { map: jest.fn() },
    onMove: jest.fn(),
  };

  const root = shallow(<PermissionsList {...props} />);

  it('renders the container of the permissions list', () => {
    expect(root.find('Paper')).toMatchSnapshot();
  });
});
