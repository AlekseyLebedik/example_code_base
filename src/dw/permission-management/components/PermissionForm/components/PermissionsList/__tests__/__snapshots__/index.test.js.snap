// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Permissions List component renders the container of the permissions list 1`] = `null`;
